azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -225,11 +226,10 @@ def get_vmagent_update_status(self):
         """
         This function gets the VMAgent update status as per the last attempted update.
         Returns: None if fail to report or update never attempted with rsm version specified in GS
-        Note: We send the status regardless of updater type. Since we call this main loop, want to avoid fetching agent family to decide and send only if
-        vm enabled for rsm updates.
+        Note: We report the status only when vm enrolled into RSM
         """
         try:
-            if conf.get_enable_ga_versioning():
+            if self.get_current_update_mode() == UpdateMode.RSM:
```

Comment:
I see that the None value returned @line 244 gets propagated all the way to 

```
def ga_status_to_v1(ga_status):
    v1_ga_status = {
        "version": ga_status.version,
        "status": ga_status.status,
        "formattedMessage": __get_formatted_msg_for_status_reporting(ga_status.message)
    }

    if ga_status.update_status is not None:
        v1_ga_status["updateStatus"] = get_ga_update_status_to_v1(ga_status.update_status)

```
Would it be better to just not report an "updateStatus", or does CRP requires the null value?

Reply:
Yes, that's what we do right. We don't add updatestatus in the payload.